crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -142,8 +144,41 @@ impl RoomList {
         let stream = stream! {
             loop {
                 let filter_fn = filter_fn_cell.take().await;
-                let (values, stream) = list
-                    .room_list_stream()
+                let (raw_values, mut raw_stream) = list.room_list_stream();
+                let mut raw_current_values = raw_values.clone();
+                let client = client.clone();
+                let raw_stream_with_recv = stream! {
+                    loop {
```

Comment:
That's the second `loop` inside a `loop` :-) Let's extract all this new stream code to another function, please :-)